### ISSUE TITLE
round trip test case

### ISSUE THREAD

**Issue by OR13** [kelseykm/vcard4#21]
Thanks for this awesome library.

it would be nice to have a round trip test case in the documentation, something like this:

```ts
  const card = new VCARD([ ... ]);
  fs.writeFileSync('example-card.vcf', card.repr()); // to xml
  const serializedCard = fs.readFileSync('example-card.vcf').toString() // from xml
  const card2 = VCARD.from(serializedCard);
  // expect card2 to equal card 1
```

As a developer, I want to be assured that all properties that are imported are preserved, then I want to add a property and export.

This seems possible with the existing parse API, but its not intuitive or obvious how exactly to do this.

Sorry if I missed this in the docs or tests.

**Comment by kelseykm** [comment]
Hello @OR13

I think you're asking about adding properties to an already constituted vCard, whether it already exists on disk or as an instance of `VCARD`.

**TL;DR**: The answer is **no**.

### Explanation for the first case (vCard already existing on disk):

The vcard4 library prioritizes adherence to vCard RFC specifications (6350, etc.). All vCards generated by this library strictly follow these specifications.

Once a vCard leaves the production system or if it was generated using other technology, its adherence to the RFCs can no longer be guaranteed by vcard4. Hence, there isn't an API to introduce externally generated vCards into the vcard4 production system.

To add a property to an existing vCard, you need to recreate the vCard using the mechanisms provided by vcard4 and then add the desired property.

Note: The `parse` function does not interact with the classes used in vCard generation. It simply parses vCards, whether created by vcard4 or otherwise, without guaranteeing specification fidelity.

### Explanation for the second case (vCard is an instance of `VCARD`):

Instances are frozen on instantiation; thus, nothing can be added or removed from them. If you want to add a property, go back a step and add it before instantiation. 


**If I misunderstood your question, please provide further clarification below.**

**Comment by OR13** [comment]
Thanks for your quick reply, the API exposed makes more sense with this context.

> To add a property to an existing vCard, you need to recreate the vCard using the mechanisms provided by vcard4 and then add the desired property.

I plan to use the output of parse to accomplish this, I'm not sure if it will be possible to keep things losseless based on your comment, but I will try.

I'm going to close this issue, since you have answered my question.
